### PR TITLE
feat: external metrics

### PR DESCRIPTION
**Pull request by Arij-Aladel**
Metrics from external modules could be used in configs.

### CODE REVIEW

**Review by IgnatovFedor — CHANGES_REQUESTED** [review]
All changes except ones in `get_metric_be_name` should be reverted. Changing only `get_metric_by_name` allows to add, for example, `sklearn.metrics:accuracy_score` to metrics list in `insults_kaggle_bert`.

Also ability to add parameters to metrics should be added. Example:
```
      {
        "name": "sklearn.metrics:accuracy_score",
        "alias": "sklearn_accuracy",
        "inputs": ["y", "y_pred_labels"],
        "normalize": false
      },
```

File: deeppavlov/core/trainers/utils.py [hunk 1]
```diff
@@ -31,14 +31,16 @@ def parse_metrics(metrics: Iterable[Union[str, dict]], in_y: List[str], out_vars
 
         metric_name = metric['name']
         alias = metric.get('alias', metric_name)
+        kwargs = {k:metric[k] for k in set(metric)-{metric_name,alias}}
```
Comment (IgnatovFedor) [inline deeppavlov/core/trainers/utils.py:34]: You forgot about possible `inputs` key. Also kwargs is completely redundant: you can use `metric` dict.
```
metric_name = metric.pop('name')
alias = metric.pop('alias', metric_name)
```
lower inputs also can be assigned with `dict.pop` instead of `dict.get`. After that `metric` dictionary either will be empty or it will contain only parameters required by metric function.

**Review by IgnatovFedor — CHANGES_REQUESTED** [review]

File: deeppavlov/core/common/metrics_registry.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
 log = getLogger(__name__)
 
+
```
[the remark this answers is not in the document]
Comment (IgnatovFedor) [reply deeppavlov/core/common/metrics_registry.py:25]: still needs to be removed)

File: deeppavlov/core/trainers/utils.py [hunk 3]
```diff
@@ -31,14 +31,16 @@ def parse_metrics(metrics: Iterable[Union[str, dict]], in_y: List[str], out_vars
...
 
         f = get_metric_by_name(metric_name)
 
         inputs = metric.get('inputs', in_y + out_vars)
         if isinstance(inputs, str):
             inputs = [inputs]
 
-        metrics_functions.append(Metric(metric_name, f, inputs, alias))
+        metrics_functions.append(Metric(metric_name, partial(f,**kwargs), inputs, alias))
+
```
[the remark this answers is not in the document]
Comment (IgnatovFedor) [reply deeppavlov/core/trainers/utils.py:43]: still needs to be removed

File: deeppavlov/core/trainers/utils.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 from collections import OrderedDict, namedtuple
 from json import JSONEncoder
 from typing import List, Tuple, Union, Iterable
+from functools import partial
```
Comment (IgnatovFedor) [inline deeppavlov/core/trainers/utils.py:18]:
```suggestion
from functools import partial
from json import JSONEncoder
from typing import List, Tuple, Union, Iterable
```
Imports should be in alphabetic order. Pycharm has command to sort them: alt+ctr+o. VSC should have something similar as well.

File: deeppavlov/core/trainers/utils.py [hunk 5]
```diff
@@ -29,16 +30,17 @@ def parse_metrics(metrics: Iterable[Union[str, dict]], in_y: List[str], out_vars
         if isinstance(metric, str):
             metric = {'name': metric, 'alias': metric}
 
-        metric_name = metric['name']
-        alias = metric.get('alias', metric_name)
-
+        metric_name = metric.pop('name')
+        alias = metric.pop('alias', metric_name)
+
```
Comment (IgnatovFedor) [inline deeppavlov/core/trainers/utils.py:35]: redundant whitespaces

File: deeppavlov/core/trainers/utils.py [hunk 6]
```diff
@@ -29,16 +30,17 @@ def parse_metrics(metrics: Iterable[Union[str, dict]], in_y: List[str], out_vars
...
-        metric_name = metric['name']
-        alias = metric.get('alias', metric_name)
-
+        metric_name = metric.pop('name')
+        alias = metric.pop('alias', metric_name)
+        
         f = get_metric_by_name(metric_name)
-
-        inputs = metric.get('inputs', in_y + out_vars)
+
```
Comment (IgnatovFedor) [inline deeppavlov/core/trainers/utils.py:37]: redundant whitespaces

File: deeppavlov/core/trainers/utils.py [hunk 7]
```diff
@@ -29,16 +30,17 @@ def parse_metrics(metrics: Iterable[Union[str, dict]], in_y: List[str], out_vars
...
         f = get_metric_by_name(metric_name)
-
-        inputs = metric.get('inputs', in_y + out_vars)
+        
+        inputs = metric.pop('inputs', in_y + out_vars)
         if isinstance(inputs, str):
             inputs = [inputs]
 
-        metrics_functions.append(Metric(metric_name, f, inputs, alias))
+        metrics_functions.append(Metric(metric_name, partial(f,**metrics), inputs, alias))
```
Comment (IgnatovFedor) [inline deeppavlov/core/trainers/utils.py:42]: `metrics` is a list, you should use `metric`

**Review by IgnatovFedor — CHANGES_REQUESTED** [review]

**Review by IgnatovFedor — APPROVED** [review]

**Review by dilyararimovna — APPROVED** [review]

**Review by dilyararimovna — APPROVED** [review]

**Review by yurakuratov — APPROVED** [review]

**Review by IgnatovFedor — APPROVED** [review]
